src/yetibot/commands/emoji.clj

```diff
@@ -0,0 +1,131 @@
+(ns yetibot.commands.emoji
+  (:require
+    [taoensso.timbre :refer [info warn error]]
+    [yetibot.core.hooks :refer [cmd-hook]]
+    [clojure.core.memoize :as m])
+  (:import
+   [com.vdurmont.emoji EmojiManager]))
+
+(defn get-all-emojis
+  "get all emoji java objects"
+  []
+  (EmojiManager/getAll))
+
+(defn parse-all-emojis
+  "create sequence of emojis"
+  []
+  (map (fn [emoji] {:unicode (.getUnicode emoji)
+                    :description (.getDescription emoji)
+                    :aliases (.getAliases emoji)
+                    :tags (.getTags emoji)
+                    }) (get-all-emojis)))
+
+(defonce emojis (parse-all-emojis))
+
+(defn all-tags
+  "emoji tags # return all known tags"
+  {:yb/cat #{:fun :img :emoji}}
+  []
+  (-> (reduce (fn [return-set emoji]
+                (let [tag-vector (:tags emoji)
+                      tag-hash-set (set tag-vector)]
+                  (clojure.set/union return-set tag-hash-set)))
+              #{}
+              emojis)
+      sort))
+
+(defn list-emojis
+  "emoji list [-i] # list all known emojis
+
+   [-i (info)] is optional, if set will return the unicode
+   as well as a vector of its aliases"
+  {:yb/cat #{:fun :img :emoji}}
+  [{[_ info-flag] :match}]
+  (let [emoji-keys (filter identity [:unicode (if info-flag :aliases)])
+        select-values (comp vals select-keys)
+        map-fn (fn [emoji]
+                    (clojure.string/join " " (select-values emoji emoji-keys)))]
+    (map map-fn emojis)))
+
+(defn random-emoji
```

Comment:
I usually rely on piping stuff to the existing [`random`](https://github.com/devth/yetibot.core/blob/master/src/yetibot/core/commands/collections.clj#L13) command. Does this have any benefit over: `emoji list | random`?